src/test/java/edu/tamu/eider/repository/EntityCrudTest.java

```diff
@@ -0,0 +1,173 @@
+                    parameterWithName("page").description("The page of Entities to retrieve"),
+                    parameterWithName("size").description("The number of Entities per page"),
+                    parameterWithName("sort").description("The way the returned page should be sorted")
+                ),
+                responseFields(
+                    subsectionWithPath("_embedded").description("A list of the Entities on the page"),
+                    subsectionWithPath("_links").description("A list of links associated with the Enttiy collection"),
+                    subsectionWithPath("page").description("A description of the attributes of the page")
+                ),
+                ENTITY_COLLECTION_LINKS
+            ));
+    }
+
+    @Test
+    public void testGetEntity() throws Exception {
+        Entity entity = entityRepo.save(TEST_ENTITY_1);
+        this.mockMvc
+            .perform(get("/entity/{id}", entity.getId().toString())
+                .accept(MediaTypes.HAL_JSON)
+                .contentType(MediaType.APPLICATION_JSON)
+            )
+            .andExpect(status().isOk())
+            .andExpect(content().contentType(MediaTypes.HAL_JSON))
+            .andExpect(jsonPath("id").value(entity.getId().toString()))
+            .andExpect(jsonPath("url").value(TEST_ENTITY_1_URL_STRING))
+            .andExpect(jsonPath("canonicalName").value(TEST_ENTITY_1_CANONICAL_NAME))
+            .andExpect(jsonPath("notes").value(TEST_ENTITY_1_NOTES))
+            .andDo(document("get-entity",
+                pathParameters(
+                    parameterWithName("id").description("The UUID id of the Entity to be retrieved")
+                ),
+                ENTITY_RESPONSE_FIELDS_SNIPPET,
+                ENTITY_LINKS
+            ));
+    }
+
+    @Test
+    public void testCreateEntity() throws Exception {
+        this.mockMvc
+            .perform(post("/entity")
+                .header("Authorization", "Basic YWRtaW46YWRtaW4=")
+                .accept(MediaTypes.HAL_JSON)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(mapper.writeValueAsString(TEST_ENTITY_1))
+            )
+            .andExpect(status().isCreated())
+            .andExpect(content().contentType(MediaTypes.HAL_JSON))
+            .andExpect(jsonPath("url").value(TEST_ENTITY_1_URL_STRING))
+            .andExpect(jsonPath("canonicalName").value(TEST_ENTITY_1_CANONICAL_NAME))
+            .andExpect(jsonPath("notes").value(TEST_ENTITY_1_NOTES))
+            .andDo(document("create-entity",
+                ENTITY_REQUEST_FIELDS_SNIPPET,
+                ENTITY_RESPONSE_FIELDS_SNIPPET,
+                ENTITY_LINKS
+            ));
+    }
+
+    @Test
+    public void testReplaceEntity() throws Exception {
+        Entity entity = entityRepo.save(TEST_ENTITY_1);
+        this.mockMvc.perform(put("/entity/{id}", entity.getId().toString())
+            .header("Authorization", "Basic YWRtaW46YWRtaW4=")
```

Comment:
I would use:

```
import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.httpBasic;
```

`.with(httpBasic("admin", "admin")))`

Could also bring in test properties for username and password. If the username and password in test properties change tests wouldn't break then.